### ISSUE TITLE
[corgan-pr-prep] T4: Update CorGAN training example

### ISSUE THREAD

**Issue by jalengg** [jalengg/PyHealth#30]
## Spec

Create `examples/corgan_mimic3_training.py` using PyHealth 2.0 API:

\`\`\`python
from pyhealth.datasets import MIMIC3Dataset, split_by_patient
from pyhealth.models.generators.corgan import CorGAN
from pyhealth.tasks import corgan_generation_mimic3_fn

base_dataset = MIMIC3Dataset(root="/path/to/mimic3", tables=["diagnoses_icd"])
sample_dataset = base_dataset.set_task(corgan_generation_mimic3_fn)
train_dataset, val_dataset, test_dataset = split_by_patient(sample_dataset, [0.8, 0.1, 0.1])

model = CorGAN(
    dataset=sample_dataset,
    latent_dim=128,
    hidden_dim=128,
    batch_size=64,
    epochs=50,
    save_dir="./corgan_checkpoints/",
)
model.train_model(train_dataset, val_dataset)
\`\`\`

## Acceptance criteria

- No bespoke dataset classes
- No argparse or HALOConfig patterns
- Mirrors `examples/halo_mimic3_training.py` structure
- Comments explain the tables= parameter and any model-specific settings
- Spec compliance review ✅
- Code quality review ✅

**Comment by jalengg** [comment]
## Why the training example needs to be rewritten, not just updated

The current `train_corgan_45k.py` script (and earlier variants) was written as a research experiment driver — it contains argparse flags, hardcoded MIMIC-III CSV paths, manual vocabulary construction, direct numpy array handling, and a custom training loop that bypasses PyHealth entirely. It demonstrates CorGAN as a standalone research tool, not as a PyHealth model.

This is a meaningful distinction. A PyHealth training example serves a different audience: a researcher or practitioner who has already loaded their clinical dataset into a `MIMIC3Dataset` object and wants to train a generative model with minimal friction. They should be able to copy the example, substitute their MIMIC-III root path, and have a working training run in under 20 lines of code.

The old script does not serve that audience. It requires the user to have already preprocessed MIMIC-III into a specific CSV format, to know CorGAN's internal vocabulary format, and to manage checkpointing manually. That is not the PyHealth promise.

## What the new example must demonstrate

The new `examples/corgan_mimic3_training.py` must demonstrate the complete PyHealth 2.0 training workflow:

1. **Dataset loading** with `MIMIC3Dataset(root=..., tables=["diagnoses_icd"])` — showing which tables are needed
2. **Task application** with `set_task(corgan_generation_mimic3_fn)` — showing how raw patient data becomes model-ready samples
3. **Patient-level split** with `split_by_patient(sample_dataset, [0.8, 0.1, 0.1])` — not random split, because generative models must not leak patient data between train and test
4. **Model instantiation** with `CorGAN(dataset=sample_dataset, ...)` — showing the key hyperparameters
5. **Training** with `model.train_model(train_dataset, val_dataset)` — not `fit()`

Each of these five steps should be separated with a comment explaining what it does and why that step exists in the workflow.

## Why split_by_patient, not random split

This is worth calling out explicitly in the example comments. If you randomly split samples (not patients), the same patient's visits can appear in both train and test sets. For generative models, this means the model can memorize specific patient trajectories and achieve low loss without learning anything generalizable. Patient-level split is not optional — it is a methodological requirement for evaluating generative EHR models.

## What the example should NOT contain

- No argparse. Configuration should be visible as Python variables, not CLI flags. The example is documentation, not a production script.
- No hardcoded result directories like `"/u/jalenj4/corgan_results/"`. Use relative paths or `"./corgan_checkpoints/"`.
- No try/except around the training call. If training fails, the error should propagate so the user sees what went wrong.
- No import of `train_corgan_45k.py` utilities. The example must be self-contained.

## Connection to the HALO precedent

`examples/halo_mimic3_training.py` (committed in the HALO integration, `halo-pr-528`) set the template: 20–30 lines, five clearly commented sections, no configuration management. The CorGAN training example should mirror that structure so that a user who has already read the HALO example immediately recognizes the pattern.

## A note on epochs

CorGAN training is slow. The example should use a realistic default (e.g., `epochs=50`) but add a comment like `# Reduce for quick testing; 50+ recommended for quality synthetic data`. This sets appropriate expectations without making the example look like it should run in 30 seconds.

**Comment by jalengg** [comment]
## T4 Complete ✅

**What was implemented:**
- New file: `examples/corgan_mimic3_training.py` (35 lines)
- Follows PyHealth 2.0 five-section pattern: dataset load → task application → train/val split → model init → train
- Uses `MIMIC3Dataset` → `corgan_generation_mimic3_fn` → `split_by_patient` → `CorGAN(dataset)` → `train_model()`
- Includes inline comments: data leakage warning on split, epoch tuning guidance
- No argparse — deliberately simple for readability as a tutorial example
- No hardcoded absolute paths

**Commit:** `2bd1bca6`

**Reviews:** Spec ✅ | Code quality ✅
